lib/s3routes/routesUtils.ts

```diff
@@ -445,23 +444,33 @@ function retrieveData(
                 });
                 // errors on server side with readable stream
                 readable.on('error', err => {
-                    log.error('error piping data from source');
-                    _destroyResponse();
+                    log.error('error piping data from source', { error: err });
                     return cbOnce(err);
                 });
                 currentStream = readable;
-                return readable.pipe(response, { end: false });
+                return readable.pipe(response, { end: false })
+                    .on('error', pipeErr => {
+                        log.error('error in pipe operation', { error: pipeErr });
```

Comment:
is there actually cases where we go (and need to go) there: I expect piping would fail either when reading from `readable` (already handled on line 446) or failing to write (handling in line 398) ?

Reply:
I added this because in case of response error we had no error handling (we have readable "error" and "end", but only "end" for response... We have it for Azure though...).
Maybe better to have it separately then

Reply:
A specific response.on('error' handling logic was added, which thus let this above line unchanged.